### DISCUSSION TITLE
Support for passkeys

### DISCUSSION THREAD

**Discussion by ptr1120** [keycloak/keycloak#16201] [Q&A] [answered]
Hello I did not find anything mentioned about status/plans regarding support of [passkey](https://www.passkeys.io/)  in Keycloak. Can someone bring light into this topic?

**Comment by DAHAG-ArisNourbakhsh** [comment, marked as answer]
I may be ignorant but isn't that already in keycloak through its support of webauthn?

**Comment by ptr1120** [comment]
You are right, I know there is full support for WebAuthn. But I am unsure whether there is some requirement for passkey as it is something on top of WebAuthn. But maybe this on top thing is only a browser or Os-specific requirement/feature and it is not relevant for Keycloak....

**Comment by ptr1120** [comment]
I created the passwordless Browser login flow as described [here](https://www.keycloak.org/docs/latest/server_admin/#creating-a-password-less-browser-login-flow). 
As a result, using Chrome browser I can use "Windows Hello", an external security key, or an external mobile device (android) for authentication.
So support of passkey in Keycloak seems fully functional.

**Reply by moritzschmitz-oviva** [reply]
This only works in conjunction with the username form though. Without using the username form, so far I only got Chrome working as a passkey. YubiKey and Android didn't work.

**Comment by wadahiro** [comment]
Hello,

Keycloak supports Passkey, but there is room for improvement.
[WebAuthn Conditional UI](https://github.com/w3c/webauthn/wiki/Explainer:-WebAuthn-Conditional-UI) is not yet supported.
I have created the following discussion:

https://github.com/keycloak/keycloak/discussions/15860

**Comment by thomasdarimont** [comment]
Just stumbled upon the [Keycloak Passkey tutorial](https://keycloak.ch/keycloak-tutorials/tutorial-passkey/) which describes how passkeys support with decent UX can be implemented with current Keycloak.

The code of the extensions from the example can be found here: https://github.com/inventage/keycloak-custom/tree/tutorial-passkey

**Reply by thomasdarimont** [reply]
@tnorimat  @wadahiro @stianst do you think it would make sense to adopt the extensions outlined in the Keycloak Passkey tutorial mentioned above? I think with a few adjustments this could be integrated into Keycloak.

Given the broad adoption of Keycloak, the native support for passkeys with good UX could give the passkeys / password-less topic quite a boost :)

**Reply by tnorimat** [reply]
@thomasdarimont I think that the native support for passkeys to keycloak is a good option. It seems that there are many actual use cases requiring passkeys for password-less solution.

**Reply by embesozzi** [reply]
Here are some UX guidelines for Passkey to have in mind: https://fidoalliance.org/ux-guidelines-for-passkey-creation-and-sign-ins

**Reply by Jamstah** [reply]
I could find an issue, is that the right next step here?

**Reply by mposolda** [reply]
There is an issue created for passkeys support: https://github.com/keycloak/keycloak/issues/23656
